src/tiled/snaphelper.cpp

```diff
@@ -41,6 +43,7 @@ void SnapHelper::toggleSnap()
 {
     mSnapToGrid = !mSnapToGrid;
     mSnapToFineGrid = false;
+    mSnapToPixels = false;
```

Comment:
I think we shouldn't turn off pixel snapping with the modifier. More likely pixel snapping is something that will stay enabled or disabled for any given project, so turning it off because somebody temporarily wants to suppress tile snapping is probably not expected in general.